### ISSUE TITLE
Instantiating twisted.trial.unittest.TestCase with no arguments raises nasty exception

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1871]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1871|
|Type|defect|
|Created|2006-06-27 17:53:13Z|

```methodName``` defaults to ```None``` but this doesn't seem to actually work.  It probably shouldn't have any default at all, or else it should be possible to instantiate a TestCase with a ```None``` ```methodName```.

<details><summary>Searchable metadata</summary>

```
trac-id__1871 1871
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1151430793000000 1151430793000000
changetime__1152418446000000 1152418446000000
version__None None
owner__ 
cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|

pyunit also breaks horribly when you instantiate TestCases without a methodName.

I've made the parameter compulsory in [branches/testcase-init-1871,](branches/testcase-init-1871,) which is ready for review.

There aren't any tests, because, well, who writes tests for method signatures?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

I can't really think of a useful test to write here, either.  +1

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [17507]) TestCase's parameter is no longer optional

 * Author: jml
 * Reviewer: exarkun
 * Fixes [#1871](https://github.com/twisted/twisted/issues/1871)

TestCase used to fail badly when no methodName parameter was supplied
to its constructor.  This branch makes the parameter compulsory.  It
does technically break API compatibility with pyunit, but not in any
meaningful way.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
